### PR TITLE
fix(chaining): lay the logic graph out in MITRE-tactic columns (#7218)

### LINKED ISSUE #7218: fix(chaining): logic view lost its per-MITRE-tactic column layout (regressed in #7099)
## Description

The scenario/simulation **Logic** view used to lay actions out in **columns by MITRE tactic** (one column per tactic, ordered by kill-chain phase), with each gating **event placed just to the left of the action it gates**. That layout was lost in the #7099 canvas rewrite (commit `38e730d47`, which deleted the old `chaining_flow/LogicFlow.tsx` + `TacticGroupNode`): the dependency-depth layering that replaced it puts the whole chain on a single alternating `action → event → action` row, which is unreadable for anything but a trivial chain.

## Fix (frontend only, layout of the hand-rolled canvas)

Restore the tactic reading in `buildLogicGraphLayout`:
- **Columns by tactic**: actions grouped into one column per MITRE tactic, columns ordered by kill-chain phase; actions top-aligned within a column.
- **Events left of their action**: each gating event sits in a lane immediately to the left of its action's column, aligned to the action's center (horizontal `event → action` arrow); events sharing a lane de-overlap; orphan events (gating nothing) fall to the far-left lane.
- **Coloured column bands** covering the **action column only** — only actions carry a TTP, so events sit outside any tactic band — with the tactic name as a header.
- **Fit anchoring**: a tall column layout no longer scrolls its column headers off-screen; the fit anchors top-left when the graph exceeds the viewport (on load and on auto-organize).

The edge set (inferred + real, back-edge pruned) is unchanged — only node positioning changed. The tactic data already flows through `ActionMeta` / `buildTacticForStep`; no backend change.

### PR DESCRIPTION
**Pull request by Seb-MIGUEL**
## Summary

The **Logic** view lost its per-MITRE-tactic column layout in the #7099 canvas rewrite: the dependency-depth layering that replaced it puts the whole chain on a single alternating `action -> event -> action` row, which is unreadable beyond a trivial chain.

This restores the tactic reading in `buildLogicGraphLayout` (layout-only; the edge set is unchanged):

- **Columns by tactic** - actions grouped into one column per MITRE tactic, columns ordered by kill-chain phase, actions top-aligned within a column (like a table).
- **Events left of their action** - each gating event sits in a lane immediately to the left of its action's column, aligned to the action's center (horizontal `event -> action` arrow). Events sharing a lane de-overlap by cascading down; orphan events fall to the far-left lane; an event gating actions in several columns takes the leftmost column's lane.
- **Coloured column bands** covering the **action column only** - only actions carry a TTP, so events are positioned outside any tactic band - with the tactic name as the column header. The `LogicGraphColumn` JSDoc documents exactly that geometry.
- **Fit anchoring** (`PanZoom`) - a tall column layout no longer scrolls its column headers off-screen: the fit anchors on the top-left when the graph exceeds the viewport in that axis (on load and on auto-organize); content that fully fits stays centered. `PanZoom` is only consumed by the Logic view, so no other canvas is affected.

`tacticForStep` + a tactic->phase-order map are threaded into the layout. The tactic data already flows through `ActionMeta`/`buildTacticForStep` (unknown tactics fall back to "Other" and sort after the mapped phases); no backend change.

Fixes #7218

## Test plan

- [x] `layout.test.ts` (7 tests): actions of a tactic share a column ordered by phase; each action sits inside its tactic band; each gating event is left of its action and vertically centered on it; **events are never inside a tactic band** (only actions carry a TTP); events sharing a lane cascade down without overlapping; a tactic missing from the order map sorts after the known ones; an event gating actions in several columns takes the leftmost lane; empty graph -> no columns.
- [x] `yarn check-ts` + `yarn eslint` (touched files) clean.
- [x] Manually verified on a real workflow (nmap/netexec AD scenario): columns per tactic with headers, events left of their actions outside the bands, and column headers visible after fit / auto-organize.

## Related

Third of the chaining QA PRs: #7211 (defined value + type link), #7213 (action->event link guard + event labels + remove action "+"), and this. These three touch adjacent chaining/logic code but are independent; merge order does not matter (only #7213 and this both touch `logic-graph/`, in non-overlapping ways - connect handles/labels vs. layout).

### CODE REVIEW

File: openaev-front/src/admin/components/chaining/logic/logic-graph/layout.ts [hunk 1]
```diff
@@ -48,9 +59,22 @@ export interface LogicGraphBBox {
   height: number;
 }
 
+/** One MITRE-tactic column: its header label and the coloured band its nodes sit in. */
+export interface LogicGraphColumn {
+  tactic: string;
+  /** The band rectangle (world coords): spans the event lane + the action column. */
+  x: number;
+  width: number;
```
[the remark this answers is not in the document]
Comment (SamuelHassine) [reply openaev-front/src/admin/components/chaining/logic/logic-graph/layout.ts:70]: Fixed in 67369d8b. Corrected the `LogicGraphColumn.x`/`width` JSDoc to describe the band as the action column only (`x = actionColX(col)`, `width = NODE_WIDTH`). Events carry no TTP, so they are placed in the lane to the LEFT of the band, outside any tactic - which is exactly what `buildLogicGraphLayout` does and what the `keeps events outside the tactic bands` test asserts. The doc now matches the code, so a future consumer sizing an overlay against a column band gets the action-column rectangle it actually is.

**Review by SamuelHassine — APPROVED** [review]
Reviewed the full changed files (layout.ts, LogicGraph.tsx, PanZoom.tsx, layout.test.ts), not just the diff. The tactic-column geometry is correct and well-documented, the edge-building code paths are untouched (layout-only as claimed), the PanZoom top-left anchoring is per-axis and only kicks in when the content exceeds the viewport (small graphs stay centered; PanZoom has no other consumer), and the unknown-tactic / orphan-event / multi-gate edge cases all resolve safely. Copilot's JSDoc mismatch was fixed in 67369d8 and three unit tests were added in 98e7ce6 to pin the uncovered layout branches (lane cascade de-overlap, unknown-tactic ordering, leftmost-lane selection for multi-column events). All checks are green, including codecov. LGTM.

**Comment by SamuelHassine** [comment]
## Review and fix summary

- **Independent review**: read the four changed files in full plus the layout's callers (`buildTacticForStep`, `tacticOrder` wiring in `LogicGraph`, `PanZoom` consumers). No must-fix issues found:
  - Band geometry: code, rendering, and tests all agree the coloured band covers the action column only; the stale JSDoc (flagged by Copilot) was corrected in 67369d8.
  - Edge set: verified against `main` that `buildInferredEdges` / `buildRealEdges` / back-edge removal / fan-in pruning are byte-identical - the change is layout-only as claimed.
  - Edge cases: an action with no tactic gets the "Other" fallback from `buildTacticForStep` (and the layout additionally guards with `?? ''`); a tactic missing from the order map sorts last via `?? 99` (no NaN); events sharing a lane cascade down without overlap; an event gating actions in several columns takes the leftmost lane; `PanZoom` anchors top-left per axis only when the content exceeds the viewport, so small graphs stay centered, and `AttackPathCanvas` has its own copy so no other view is affected.
- **Fixes pushed**: 98e7ce6 adds three unit tests pinning the uncovered layout branches above (7 tests total, all passing; `yarn eslint` + `yarn check-ts` clean).
- **Threads**: the single Copilot thread (JSDoc mismatch) is fixed, replied, and resolved.
- **CI**: all checks green, including `codecov/patch` (the new tests lifted the patch coverage above the earlier 64.10% failure).
- **PR body**: re-flowed to ASCII and updated to reflect the final state (7 tests, fit-anchoring scope).
